Build a regression model.

In [1]:
import numpy as np
import pandas as pd
from sklearn import linear_model, datasets

# load data
df = pd.read_csv('bike_yelp_df.csv')
x = pd.DataFrame(df['empty_slots'])
y = pd.DataFrame(df['free_bikes'])

# view linear regression
x.shape
x.head()
y.shape
y.head()

free_bikes
0          22
1          22
2          22
3          22
4          22

Provide model output and an interpretation of the results. 

In [2]:
import statsmodels.api as sm

# add constant
x = sm.add_constant(x)
model = sm.OLS(y, x).fit()
summary = model.summary()
print(summary)

                            OLS Regression Results                            
Dep. Variable:             free_bikes   R-squared:                       0.298
Model:                            OLS   Adj. R-squared:                  0.297
Method:                 Least Squares   F-statistic:                     1043.
Date:                Thu, 14 Dec 2023   Prob (F-statistic):          3.80e-191
Time:                        11:02:47   Log-Likelihood:                -7378.6
No. Observations:                2463   AIC:                         1.476e+04
Df Residuals:                    2461   BIC:                         1.477e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const          14.8453      0.204     72.835      

The p-value is very low, indicating that the model is statistically significant, but by looking at the R-squared value, it tells us that only 29.8% of the data can be explained by this model. Additionally, we can see that 'empty_slots' is a strong predictor of 'free_bikes' by looking at the coefficient values. There is a negative relationship - if there is an empty slot, your chances of getting a free bike decreases by 0.614. 